### PR TITLE
build: Check for -Wvla before use

### PR DESCRIPTION
**Pull request by ghost**
OpenBSD's GCC (4.2.1) doesn't have the -Wvla flag.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -150,9 +150,14 @@ if(MINGW)
   add_definitions(-D__USE_MINGW_ANSI_STDIO)
 endif()
 
+# OpenBSD's GCC (4.2.1) doesn't have -Wvla
+check_c_compiler_flag(-Wvla HAS_WVLA_FLAG)
+if(HAS_DIAG_COLOR_FLAG)
```
Comment (justinmk) [inline CMakeLists.txt]: HAS_WVLA_FLAG?

Comment (Pyrohh) [inline CMakeLists.txt]: ....lol

**Comment by Pyrohh** [comment]
Thanks @justinmk, re-pushed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -172,14 +177,9 @@ if(TRAVIS_CI_BUILD)
   add_definitions(-Werror)
 endif()
 
-if(CMAKE_COMPILER_IS_GNUCC)
-  check_c_compiler_flag(-Og HAS_OG_FLAG)
-else()
-  set(HAS_OG_FLAG 0)
-endif()
```
Comment (justinmk) [inline CMakeLists.txt]: @jszakmeister Did this intentionally: https://github.com/neovim/neovim/pull/3137#discussion_r38042611

Comment (Pyrohh) [inline CMakeLists.txt]: As he said, it might be a bit conservative, but the benefit is minimal so I'll remove it.

**Comment by Pyrohh** [comment]
repushed

**Comment by justinmk** [comment]
LGTM, feel free to merge after travis.

**Comment by Pyrohh** [comment]
Thanks for the review @justinmk.

**Pull request merged**